engines/ags/engine/ac/display.cpp

```diff
@@ -67,26 +67,27 @@ struct DisplayVars {
 	int fulltxtheight = 0; // total height of all the text
 } disp;
 
-// Pass yy = -1 to find Y co-ord automatically
-// allowShrink = 0 for none, 1 for leftwards, 2 for rightwards
-// pass blocking=2 to create permanent overlay
-ScreenOverlay *_display_main(int xx, int yy, int wii, const char *text, int disp_type, int usingfont,
-		int asspch, int isThought, int allowShrink, bool overlayPositionFixed, bool roomlayer) {
+// Generates a textual image and returns a disposable bitmap
+Bitmap *create_textual_image(const char *text, int asspch, int isThought,
+							 int &xx, int &yy, int &adjustedXX, int &adjustedYY, int wii, int usingfont, int allowShrink,
+							 bool &alphaChannel) {
+	//
+	// Configure the textual image
+	//
 	const bool use_speech_textwindow = (asspch < 0) && (_GP(game).options[OPT_SPEECHTYPE] >= 2);
 	const bool use_thought_gui = (isThought) && (_GP(game).options[OPT_THOUGHTGUI] > 0);
 
-	bool alphaChannel = false;
-	char todis[STD_BUFFER_SIZE];
-	snprintf(todis, STD_BUFFER_SIZE - 1, "%s", text);
+	alphaChannel = false;
 	int usingGui = -1;
 	if (use_speech_textwindow)
 		usingGui = _GP(play).speech_textwindow_gui;
 	else if (use_thought_gui)
 		usingGui = _GP(game).options[OPT_THOUGHTGUI];
 
 	int padding = get_textwindow_padding(usingGui);
```

Comment:
It's not going to break anything, but the upstream commit also change the above line to use a `const int`.

Reply:
Ah. I see you actually changed it in the next commit. 😄 
So it's fine.